src/plots/plot.ts

```diff
@@ -451,9 +451,13 @@ export class Plot extends Component {
           return;
         }
 
+        const plot = this;
         lightweightPlotEntities.push({
           datum,
-          position,
+          get position() {
+            // only calculate position when needing to improve pan zoom performance #3159
+            return plot._pixelPoint.call(plot, datum, datumIndex, dataset);
```

Comment:
just tried this out, yup we won't be able to use that in the future =(

this guy pops up:
`Uncaught TypeError: Converting circular structure to JSON`